compiler/src/iree/compiler/DispatchCreation/BubbleUpExpandShapes.cpp

```diff
@@ -117,6 +117,37 @@ struct BubbleExpandThroughExtract final
 
 } // namespace
 
+/// If the domain of the operation is being expanded by unit dimensions, check
+/// if it's possible to have an infinite loop where the unit dim expansion keeps
+/// on propagating infinitely.
+static bool canCauseReshapingLoopByExpansion(Operation *producer,
+                                             Operation *consumer) {
+  bool isExpandingToUnitDims = false;
+  if (auto expandShapeOp = dyn_cast<tensor::ExpandShapeOp>(consumer)) {
```

Comment:
I think we need the same check for collapse_shape right?

Reply:
The collapse_shape will eventually propagate up as an expand_shape, so it will stop the loop from ever happening.

The bigger reason why I only implemented it for expand_shape is that it's less common for an operation to have multiple results than an operation to have multiple inputs. So i'd rather have a conservative check for multiple results since that occurs less.